components/server/src/auth/host-context-provider-impl.ts

```diff
@@ -82,16 +82,19 @@ export class HostContextProviderImpl implements HostContextProvider {
             const { host } = config;
 
             const existingContext = this.dynamicHosts.get(host);
-            const existingConfig = existingContext && existingContext.authProvider.params;
-            if (existingConfig && config.id === existingConfig.id) {
-                if (existingConfig.host !== config.host) {
+            if (existingContext) {
+                const existingConfig = existingContext.authProvider.params;
+                const sameId = config.id === existingConfig.id;
+                const sameHost = config.host === existingConfig.host;
+                if (!sameHost || !sameId) {
                     log.warn("Ignoring host update for dynamic Auth Provider: " + host, { config, existingConfig });
                     continue;
                 }
-                if (existingConfig.status === config.status) {
-                    if (!!config.oauthRevision && existingConfig.oauthRevision === config.oauthRevision) {
-                        continue;
-                    }
+                const sameOwner = config.ownerId === existingConfig.ownerId;
+                const sameStatus = config.status === existingConfig.status;
+                const hasOAuthRevisionChanged =
+                    !!config.oauthRevision && existingConfig.oauthRevision === config.oauthRevision;
+                if (sameOwner && sameStatus && hasOAuthRevisionChanged) {
```

Comment:
Why do we jump over the fast path here (Jason.stringify), if we just checked all properties?

Reply:
good question. I don't recall, why the `JSON.stringify(existingConfig.oauth) === JSON.stringify(config.oauth)` was left in place. in theory, checking for the `sameOAuthRevision` should include it already.

Reply:
I removed the `JSON.stringify` comparison, as I see no other paths which would tamper with the config object here, i.e. it's kind of useless double-checking.

Reply:
Originally the `if (nothingChanged) { continue; }` was preferrred, and the JSON compare served as fallback.

With your last change both (accidentally) got merged.

IMO it's ok to drop the fallback, as it should not be required anymore. :+1: